Sources/MapboxNavigation/NavigationMapView+BuildingHighlighting.swift

```diff
@@ -15,13 +15,122 @@ extension NavigationMapView {
     }
     
     /**
      Removes the highlight from all buildings highlighted by `highlightBuildings(at:in3D:)`.
      */
     public func unhighlightBuildings() {
-        
+        guard let _ = try? mapView.style.getLayer(with: IdentifierString.buildingExtrusionLayer, type: FillExtrusionLayer.self).get() else { return }
+        foundBuildingIds = Set<Double>()
+        let _ = mapView.style.removeStyleLayer(forLayerId: IdentifierString.buildingExtrusionLayer)
     }
+
+    private func featureIdentifierToDouble(features: [Feature]) -> Double? {
+        if let featureIdentifier = features.first?.identifier {
+            switch featureIdentifier {
+            case .string(let stringId): return Double(stringId)
+            case .number(let numberId):
+                switch numberId {
+                case .double(let doubleId): return Double(doubleId)
+                case .int(let intId): return Double(intId)
+                }
+            }
+        }
+        return nil
+    }
+
+    private func buildingIdentifier(at coordinate: CLLocationCoordinate2D, in3D extrudesBuildings: Bool = true) {
+        let screenCoordinate = mapView.point(for: coordinate, in: self)
+        if let identifiers = try? mapView.__map.getStyleLayers().compactMap({$0.id}).filter({ $0.contains("building") }) {
+            mapView.visibleFeatures(at: screenCoordinate,
+                                    styleLayers: Set(identifiers),
+                                    completion: { [weak self] result in
+                                        guard let validSelf = self else { return }
+                                        if case .success(let features) = result {
+                                            if let identifier = self?.featureIdentifierToDouble(features: features) {
+                                                self?.foundBuildingIds.insert(identifier)
+                                            }
+                                        }
+                                    })
+        }
+    }
+
+    private func addBuildingsLayer(with identifiers: Set<Double>, in3D: Bool = false, extrudeAll: Bool = false) {
+        mapView.style.removeStyleLayer(forLayerId: IdentifierString.buildingExtrusionLayer)
+        if identifiers.isEmpty { return }
+        var highlightedBuildingsLayer = FillExtrusionLayer(id: IdentifierString.buildingExtrusionLayer)
+        highlightedBuildingsLayer.source = "composite"
+        highlightedBuildingsLayer.sourceLayer = "building"
+
+        if extrudeAll {
+            highlightedBuildingsLayer.filter = Exp(.eq) {
+                Exp(.get) {
+                    "extrude"
+                }
+                "true"
+            }
+        } else {
+            highlightedBuildingsLayer.filter = Exp(.all) {
+                Exp(.eq) {
+                    Exp(.get) {
+                        "extrude"
+                    }
+                    "true"
+                }
```

Comment:
Shall we store this expression as local variable to be able to re-use it in several places (I can see two usages)?

Reply:
Thanks for the suggestion! I'll extract these expressions as a local variable.

Reply:
Thanks for the suggestion! I'll extract these expressions as a local variable.